pkg/kubectl/cmd/templates/command_groups.go

```diff
@@ -35,6 +35,20 @@ func (g CommandGroups) Add(c *cobra.Command) {
 	}
 }
 
+// PruneCommands removes commands that are set to nil. This allows commands to disable
+// themselves by returning a nil cobra.Command.
+func (g CommandGroups) PruneCommands() {
+	for i, group := range g {
+		gc := make([]*cobra.Command, 0, len(group.Commands))
+		for _, c := range group.Commands {
+			if c != nil {
+				gc = append(gc, c)
+			}
+		}
+		g[i].Commands = gc
+	}
+}
```

Comment:
This means that `kubectl help debug` will fail to print the documentation if the feature is disabled by the environment.
I assume this is what you want, but what if it instead printed information about how to enable the command?